src/transformers/commands/add_new_model_like.py

```diff
@@ -402,12 +403,74 @@ def get_module_from_file(module_file: Union[str, os.PathLike]) -> str:
 _re_class_func = re.compile(r"^(?:class|def)\s+([^\s:\(]+)\s*(?:\(|\:)", flags=re.MULTILINE)
 
 
+def remove_attributes(obj, attrs_to_remove=None):
+    """Remove some attributes in `obj`.
+
+    The string `obj` needs to be able to be parsed by `ast.parse`.
+    """
+    if not attrs_to_remove:
```

Comment:
If possible, I would avoid to use `re` and to deal with the `indent` manually. I am a noob here.
Use `ast.parse`, I feel more comfortable 🙏 .

So far, this function can only work with things defined in a class.

Reply:
@sgugger I have run with this change. It looks good - `pipeline_model_mapping` and `is_pipeline_test_to_skip` not copied now.

Reply:
I'm really wary about using `ast` (which is not simpler than a regex/string search by the way 😅 ) as there is no guarantee of compatibility between Python versions. It's not sure the syntax will be the same for future releases or that what you experienced on your Python version will work the same on others.

Reply:
A super super search told me that the `ast` problem is when using a `ast module` of one version to parse another version of `python code`.

In our case, this is not the use case. The users' python version's ast is used to parse `transformers` code (which should work, otherwise `transformers` is (very likely) not going to work in that env. anyway).

As the parsed results, I don't have a proof, but I think they are consistent. What I mean is that the structure of `ast.Module`, `ast.ClassDef`, `ast.Assign` is not going to change across python versions of python 3.x.y. And regarding `python 4.x.y`, well, it's not going to happen until AGI arrives.